examples/federated_analysis/configs/fed_analysis/config/config_fed_client.json

```diff
@@ -6,7 +6,8 @@
       "executor": {
         "path": "analysis_executor.AnalysisExecutor",
         "args": {
-          "data_root": "PWD/data"
+          "data_root": "PWD/data",
+          "supported_task": "histogram"
```

Comment:
I want to get the concept across that this executor runs for a specific task defined in the ["tasks" list](https://github.com/NVIDIA/NVFlare/blob/5b893c7a480a7db16c732412af4440b4c1b76914/examples/federated_analysis/configs/fed_analysis/config/config_fed_client.json#L5). First, I had `supported_task_histogram` but that felt too long. Right now, the executor only supports one task, so just `supported_task` seems sufficient.

Reply:
I like the use of `supported_task` here to make it clear that an executor supports a specific task that is assigned by the controller workflow, whereas the controller may assign multiple tasks.

Question: How do we handle an executor that supports multiple tasks (which is possible, right)?  Would we just repeat the executor block, one for each `supported_task`?  Or would we use a single executor block with a list of supported tasks?  It's not clear how this maps back to the implementation of the executor or how this `supported_task` argument is used.

Reply:
It's like in this [example](https://github.com/NVIDIA/NVFlare/blob/f360fba591941989a59d30f7be3477e635433a48/examples/hello-pt/custom/cifar10trainer.py#L114). The execute call just needs to decide which task to execute.

Reply:
That makes sense in the executor implementation.  I'm more wondering what this looks like in the client config.  In the [config](https://github.com/NVIDIA/NVFlare/blob/f360fba591941989a59d30f7be3477e635433a48/examples/hello-pt/config/config_fed_client.json) for that example, we define the list of tasks,`["train", "submit_model"]` implemented by the `cifar10trainer.Cifar10Trainer` executor.

Does `supported_tasks` then also become a list?  Or do we need a unique definition for each supported task so we can handle them separately, in which case it might make more sense to use `"supported_task_<taskname>": "<taskname>" as you did initially with `supported_task_historgram`.

Reply:
Yes, once you have more than one task supported by an executor, it makes sense to give them separate arguments like `supported_task_<taskname>`. The task list inside the client config is only used by the nvflare system to run the execute function of executors matching the requested task. The Executor implementation should still check that the correct task is being executed.